### PR TITLE
Integrate with Helix vm freeze event handler

### PR DESCRIPTION
**Pull request by JingQianCloud**
1. Bump up Helix core from 1.0.1 to 1.0.3.
2. For frontend, integrate with Helix VM event handler.

### CODE REVIEW

File: ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixParticipant.java [hunk 1]
```diff
@@ -95,7 +101,31 @@ public HelixParticipant(ClusterMapConfig clusterMapConfig, HelixFactory helixFac
...
+    // Cloud Event Callback Property: A property to define the maintenance event handling behavior for a Helix manager.
+    // User can specify which callback implementation class they want to use for Helix operations, if not specified, it will use org.apache.helix.cloud.event.helix.DefaultCloudEventCallbackImpl.class by default
+    CloudEventCallbackProperty callbackProperty = new CloudEventCallbackProperty(Collections.emptyMap());
+
+    // Enable/disable Helix optional operations for a Helix manager via CloudEventCallbackProperty.
+    callbackProperty.setHelixOperationEnabled(HelixOperation.ENABLE_DISABLE_INSTANCE, true);
+    callbackProperty.setHelixOperationEnabled(HelixOperation.MAINTENANCE_MODE, true);
+
+    // Helix Cloud Property: General cloud-related property for a Helix manager, includes CloudEventCallbackProperty.
+    HelixCloudProperty cloudProperty = new HelixCloudProperty(new CloudConfig(new ZNRecord(clusterName)));
```
Comment (mgao0) [inline ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixParticipant.java:118]: As we discussed offline, there are two ways to supply a cloud config, you could build one in code, or read from zk in ZKHelixManager constructor. `new CloudConfig(new ZNRecord(clusterName)` this means a default cloud config which is usually used when cloud is not enabled. When on cloud, if the cloud config is to be read from zk, then it's fine; but if a cloud config is meant to be built in code, then you will need to do more here.

**Review by mgao0 — COMMENTED** [review]
Generally looks good to me. Left one comment to consider. Discussed offline about having listeners registered for non-cloud environment won't do harm because no signal is going to trigger the listeners.

File: gradle/dependency-versions.gradle [hunk 2]
```diff
@@ -19,7 +19,7 @@ ext {
     javaxVersion = "3.0.1"
     nettyVersion = "4.1.74.Final"
     nettyTcnativeVersion = "2.0.49.Final"
-    helixVersion = "1.0.1"
```
Comment (xyuanlu) [inline gradle/dependency-versions.gradle:22]: Please don't use 1.0.3. This version will be deprecated.

Comment (JingQianCloud) [reply gradle/dependency-versions.gradle:22]: Ack. Will wait for the next Helix release.

**Comment by JingQianCloud** [comment]
Discard the PR.

**Pull request closed without merging**